src/server.py

```diff
@@ -464,37 +420,97 @@ def run_seta_details_query():
     if (str(branch) in ['fx-team', 'mozilla-inbound', 'autoland']) is not True \
             and str(branch) != '':
         abort(404)
-    for d in query:
-        jobtype.append([d[0], d[1], d[2]])
 
-    # We call valve to determine what kind of jobs we should return only if
-    # this request is comes from taskcluster. Otherwise, we just return what people
-    # request for.
+    alljobs = JOBSDATA.jobtype_query()
+
+    # For the case of TaskCluster request, we don't care which priority the user request.
+    # We return jobs depend on the strategy that we return high value jobs as default and
+    # return all jobs for every 5 push or 90 minutes for that branch.
     if request.headers.get('User-Agent', '') == 'TaskCluster':
 
+        # we query all jobs rather than jobs filter by the requested priority in here,
+        # Because we need to set the job returning strategy depend on different job priority.
+        query = session.query(JobPriorities.platform,
+                              JobPriorities.buildtype,
+                              JobPriorities.testtype,
+                              JobPriorities.priority,
+                              JobPriorities.timeout
+                              ).all()
+
         # We should return full job list as a fallback, if it's a request from
         # taskcluster and without head_rev or pushlog_id in there
-        if head_rev or pushlog_id:
-            priority = valve(head_rev, pushlog_id, branch, priority)
+        branch_info = session.query(TaskRequests.counter, TaskRequests.datetime).filter(
+            TaskRequests.branch == branch).all()
+
+        time_of_now = datetime.datetime.now()
+        time_of_request = time.strftime("%Y-%m-%d %H:%M:%S")
+
+        # If we got nothing related with that branch, we should create it.
+        if len(branch_info) == 0:
+            # time_of_lastreset is not a good name anyway :(
+            # And we treat all branches' reset_delta is 90 seconds, we should find a
+            # better delta for them in the further.
+            branch_data = TaskRequests(str(branch), 1, str(time_of_request), 5400)
+            try:
+                session.add(branch_data)
+                session.commit()
+            except Exception:
+                session.rollback()
+
+            finally:
+                session.close()
+            counter, string_time, reset_delta = 1, time_of_request, 5400
```

Comment:
I don't like this style of assignment, it is a neat shortcut, but potential for confusion.  Also 5400 is 90 minutes (not seconds), which looks correct- but lets make it a constant, even if it is in the function.